Sd 979 hente filer fra henvendelse

nils-arne opened the pull request.

nils-arne committed:
Implementert første versjon av workflow for preprod-default. Ryddet opp i nais*.yml filene. Skrevet om innlesing av miljøparametere for også å kunne hente fra Vault når deploy til nais. Lagt om til å bruke embedded postgres ved lokal testing. Lagt inn kode for å refreshe tilgang til postgres server, ved å hente nye tilganger fra Vault ved jevne mellomrom.

nils-arne committed:
Fikset feil på angivelse av cluster, skal være dev-fss

nils-arne committed:
Hevet spring boot versjon og lagt inn sikkerhet

nils-arne committed:
Lagt inn logging for debugging av problem med pålogging til database

nils-arne committed:
Fjernet ubrukte variable, fjernet overflødig "/"

nils-arne committed:
Endret databasehost

nils-arne committed:
Oppgradert flyway versjon

nils-arne committed:
Ryddet opp i yml filene etter testing

sjoblomjsqueed commented on src/main/kotlin/no/nav/soknad/arkivering/soknadsfillager/service/HentFilerService.kt:
```diff
@@ -1,13 +1,16 @@
-class HentFilerService(private val filRepository: FilRepository) {
-	private val logger = LoggerFactory.getLogger(javaClass)
+class HentFilerService(private val filRepository: FilRepository, private val henvendelse: HenvendelseInterface, private val appConfig: AppConfiguration) {
+	private val logger = LoggerFactory.getLogger(HentFilerService::class.java)
```
Trenger ikke å angi klassen :)
`private val logger = LoggerFactory.getLogger(javaClass)`

sjoblomjsqueed commented on src/main/kotlin/no/nav/soknad/arkivering/soknadsfillager/db/VaultCredentialService.kt:
```diff
@@ -0,0 +1,65 @@
+import no.nav.soknad.arkivering.soknadsfillager.ApplicationState
+import org.slf4j.LoggerFactory
+
+private val log = LoggerFactory.getLogger("no.nav.soknad.arkivering.soknadsfillager.db")
```
Trenger ikke å angi klasspathen :)
`private val log = LoggerFactory.getLogger(javaClass)`

nils-arne replied:
Flyttet log variabelen inn i class og skrev om.

sjoblomjsqueed commented on src/main/kotlin/no/nav/soknad/arkivering/soknadsfillager/henvendelse/RestClient.kt:
```diff
@@ -0,0 +1,87 @@
+import reactor.netty.tcp.TcpClient
+
+@Service
+class RestClient(private val appConfig: AppConfiguration): HenvendelseInterface {
```
Ett mer beskrivande namn på klassen kanske vore HenvendelseClient?

sturleh replied:
jeg tenkte at det var bedre å tenke på hva den faktisk gjør.
HentFilerFraHenvendelseClient

sjoblomjsqueed replied:
Det är inte vad den faktiskt gör. Gränssnittet har två metoder, en för att hente filer och en för att slette.

nils-arne replied:
Endret til HenvendelseClient

sjoblomjsqueed commented on src/main/kotlin/no/nav/soknad/arkivering/soknadsfillager/henvendelse/RestClient.kt:
```diff
@@ -0,0 +1,87 @@
+@Service
+class RestClient(private val appConfig: AppConfiguration): HenvendelseInterface {
+
+	private val logger = LoggerFactory.getLogger(RestClient::class.java)
```
Trenger ikke å angi klassnamnet :)
`private val logger = LoggerFactory.getLogger(javaClass)`

sturleh commented on src/main/kotlin/no/nav/soknad/arkivering/soknadsfillager/repository/FilDbData.kt:
```diff
@@ -1,13 +1,14 @@
 @Entity
 @Table(name = "documents")
-data class FilDbData(@Id @Column(name = "id") val uuid: String, @Column(name = "data", columnDefinition = "bytea") val data: ByteArray?) {
+data class FilDbData(@Id @Column(name = "id") val uuid: String, @Column(name = "data", columnDefinition = "bytea") val data: ByteArray?, @Column(name = "opprettet", columnDefinition = "TIMESTAMP WITH TIME ZONE") val opprettet: LocalDateTime?) {
```
burde det ikke het fil og ikke data?

sjoblomjsqueed replied:
Enig

nils-arne replied:
For å endre kolonnenavn har jeg implementert migreringsskript også. 
Det som jeg ikke forstår i implementasjonen av FilDbData er equals metoden. Det kan da ikke være riktig. Jeg tror at eneste kriterium for likhet er at uuid er lik, ikke at det er samme fil.

sturleh commented on src/main/kotlin/no/nav/soknad/arkivering/soknadsfillager/repository/FilDbData.kt:
```diff
@@ -1,13 +1,14 @@
 import javax.persistence.Table
 
 @Entity
 @Table(name = "documents")
```
norks eller engelsk? vi er liksom midt mellom alt..

nils-arne replied:
Går for engelske navn på tabell og kolonner.

sturleh commented on src/main/resources/db/migration/V1.0__initial_tables.sql:
```diff
@@ -0,0 +1,7 @@
+CREATE TABLE documents
+(
+    id        VARCHAR(255)  NOT NULL,
+    data      BYTEA,
```
burde det ikke vært fil og ikke data

sjoblomjsqueed replied:
Enig

nils-arne replied:
Har laget nytt script som bytter data til document

sturleh reviewed (commented):
foreslår en refakturering her etter pr er godkjent. Navgivning norsk/engelsk bør vi snakke om.

nils-arne committed:
Fjernet overflødig configurasjon i application.yml
Endret log level i Vault.kt. Fikset håndtering av applicationState parameteren.

nils-arne committed:
Litt navneendring og fiksing av feil på setting av delt passord

nils-arne committed:
Merge remote-tracking branch 'origin/master' into sd-979-hente-filer-fra-henvendelse

# Conflicts:
#	pom.xml

nils-arne committed:
Databasebruker skal fornyes ved jevne mellomrom. Forsøker med omskriving av initiering av datasource.

nils-arne committed:
Skrevet om oppstart av embeddedPostgres mer lik som kjøring på nais for lettere å debugge.

nils-arne committed:
Fikset kompileringsfeil

nils-arne commented on pom.xml:
```diff
@@ -17,39 +17,89 @@
+		</dependency>
+		<dependency>
+			<groupId>com.opentable.components</groupId>
+			<artifactId>otj-pg-embedded</artifactId>
```
Denne er trukket inn for å starte embeddedpostgres som vi benytter både for test og kjøring via spring og docker.

nils-arne commented on src/main/kotlin/no/nav/soknad/arkivering/soknadsfillager/db/Vault.kt:
```diff
@@ -0,0 +1,63 @@
+import no.nav.soknad.arkivering.soknadsfillager.ApplicationState
+import org.slf4j.LoggerFactory
+
+private val log = LoggerFactory.getLogger("no.nav.soknad.arkivering.soknadsfillager.vault")
```
Kan bare benyttes inne i class definisjoner.

nils-arne committed:
Opprydding etter PR
